### PR TITLE
[hdf5] add tools and fortran feature

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

File: ports/hdf5/portfile.cmake [hunk 1]
```diff
@@ -40,26 +54,50 @@ vcpkg_configure_cmake(
...
 
-#Linux build create additional scripts here. I dont know what they are doing so I am deleting them and hope for the best
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
-endif()
-
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
+file(READ "${CURRENT_PACKAGES_DIR}/share/hdf5/hdf5-config.cmake" contents)
```
Comment (JackBoosY) [inline ports/hdf5/portfile.cmake]: We already have a new function named `vcpkg_replace_string` to solve this situation.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/hdf5/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/hdf5/portfile.cmake]:
```
function(vcpkg_replace_string filename match_string replace_string)
    file(READ ${filename} _contents)
    string(REPLACE "${match_string}" "${replace_string}" _contents "${_contents}")
    file(WRITE ${filename} "${_contents}")
endfunction()
```

does not support `[[literalstring]]` syntax due to the added quotes around `match_string` and `replace_string` and I am not to eager to try out how many extra `\` i need to escape the string correctly

Comment (Neumann-A) [reply ports/hdf5/portfile.cmake]: also the name of the function is misleading. It should be `vcpkg_replace_string_in_file` and should also support `REGEX REPLACE`

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
Is this a WIP PR? Or are we going to merge it with the feature unsupported?

**Comment by Neumann-A** [comment]
>Is this a WIP PR? Or are we going to merge it with the feature unsupported?

I consider this PR finished.
The feature is only unsupported since a newer ninja is required and due to the fact that on windows vcpkg does not have an automatic fortran compiler download. If you manually setup everything it should work.

**Comment by strega-nil** [comment]
Cool! We'll discuss this at our next PR meeting :)

File: ports/hdf5/CONTROL [hunk 2]
```diff
@@ -12,10 +12,18 @@ Build-Depends: mpi
...
 Feature: szip
 Description: Build with szip
 Build-Depends: szip
 
 Feature: zlib
 Description: Build with zlib
 Build-Depends: zlib
+
+Feature: fortran
+Description: Build with fortran (not implemented yet)
```
Comment (BillyONeal) [inline ports/hdf5/CONTROL]: This should either work, in which case it shouldn't say (not implemented yet), or it doesn't work, in which case this isn't ready to merge. Maybe this should link to a description of the prerequisites or something?

Comment (Neumann-A) [reply ports/hdf5/CONTROL]: Better?

**Review by BillyONeal — APPROVED** [review]

**Review by dan-shaw — APPROVED** [review]

**Pull request merged**